src/event-manager/index.ts

```diff
@@ -0,0 +1,79 @@
+import { EventEmitter } from 'events';
+import Ain from '../ain';
+import {
+  BlockchainEventTypes,
+  BlockFinalizedEventConfig,
+  EventChannelConnectionOption,
+  EventConfigType,
+  ValueChangedEventConfig,
+} from '../types';
+import EventFilter from './event-filter';
+import EventChannelClient from './event-channel-client';
+import { PushId } from '../ain-db/push-id';
+import EventCallbackManager from './event-callback-manager';
+
+export default class EventManager {
+  private _ain: Ain;
+  private readonly _filters: { [filterId: string]: EventFilter };
+  private readonly _eventEmitters: { [filterId: string]: EventEmitter };
+  private readonly _eventCallbackManager: EventCallbackManager;
+  private readonly _eventChannelClient: EventChannelClient;
+
+  constructor(ain: Ain) {
+    this._filters = {};
+    this._eventEmitters = {};
+    this._ain = ain;
+    this._eventCallbackManager = new EventCallbackManager(this);
+    this._eventChannelClient = new EventChannelClient(ain, this._eventCallbackManager);
+  }
+
+  async connect(connectionOption?: EventChannelConnectionOption) {
+    await this._eventChannelClient.connect(connectionOption || {});
+  }
+
+   getEventEmitter(filterId: string) {
+    return this._eventEmitters[filterId];
+  }
+
+  disconnect() {
+    this._eventChannelClient.disconnect();
+  }
+
+  buildFilterId() {
+    return PushId.generate();
+  }
+
+  subscribe(
+      eventType: 'BLOCK_FINALIZED',
+      config: BlockFinalizedEventConfig): EventEmitter;
+  subscribe(
+      eventType: 'VALUE_CHANGED',
+      config: ValueChangedEventConfig): EventEmitter;
+  subscribe(eventTypeStr: string, config: EventConfigType): EventEmitter {
```

Comment:
could you also add 'unsubscribe'?
If the protocol message or the server side is not ready yet, could you add at least the function prototype?

Reply:
According to meeting, let's add `ubsubscribe` function to `subscription`!